sympy/core/exprtools.py

```diff
@@ -25,6 +25,23 @@
 _eps = Dummy(positive=True)
 
 
+def is_int(x):
```

Comment:
I think I would expect a function of this name to return true if x is an instance of `int` which is not really the intention here.

How about `integer_valued` by analogy with `equal_valued`. Also maybe this belongs in core.numbers along with `equal_valued`.